python/lsst/summit/utils/guiders/plotting.py

```diff
@@ -351,81 +431,98 @@ def _starMosaic(
-            else:
-                shape = self.guiderData[detName, 0].shape
-                refCenter = (float(shape[1] // 2), float(shape[0] // 2))
-                centroidOffset = (0.0, 0.0)
-
-            # Render the stamp panel
-            imObj, centerCutout, shape, _ = renderStampPanel(
-                axs[detName],
+            ax = axs[detName]
+
+            # Get star info for this detector
+            starInfo = self._getStarInfo(detName, stampNum)
+
+            # Render the stamp
+            imObj = renderStampPanel(
+                ax,
                 self.guiderData,
                 detName,
                 stampNum,
-                center=refCenter,
+                viewCenter=starInfo.refCenter,
                 cutoutSize=cutoutSize,
                 plo=plo,
                 phi=phi,
-                annotate=False,
             )
-            # Static overlays (when not animating)
-            if not isAnimated:
-                addStaticOverlays(axs[detName], detName, centerCutout, cutoutSize, camRotAngle=camAngle)
-
             artists.append(imObj)
 
-            if withStars:
-                starCross = plotStarCentroid(
-                    axs[detName],
-                    centerCutout,
-                    deltaXY=centroidOffset,
-                    markerSize=8 if cutoutSize > 0 else 0,
+            # Static overlays (reference frame elements)
+            if not isAnimated:
+                addReferenceOverlays(ax, detName, starInfo.refCenter, cutoutSize)
+
+            # Animated overlays (star position elements)
+            if starInfo.hasData and cutoutSize > 0:
+                # Rotated crosshair follows the star
+                crosshairArtists = plotCrosshairRotated(
+                    starInfo.starCenter,
+                    90 + camAngle,
+                    ax,
+                    color="grey",
+                    size=cutoutSize,
                 )
-                jitter = getStdCentroid(self.starsDf, self.expId)
+                artists.extend(crosshairArtists)
 
+                # Star centroid marker
+                starCross = plotStarCentroid(
+                    ax,
+                    starInfo.starCenter,
+                    markerSize=8,
+                )
                 artists.extend(starCross)
 
-            cutoutShapeList.append(shape)
+                jitter = getStdCentroid(self.starsDf, self.expId)
 
+        # Center panel annotation
         stampInfo = annotateStampInfo(
             axs["center"], expid=self.expId, stampNum=stampNum, nStamps=nStamps, view=view, jitter=jitter
         )
         artists.append(stampInfo)
 
-        cutoutSize = np.max(cutoutShapeList) if cutoutShapeList else 30
-
+        # Arrow panel
         if not isAnimated:
-            drawArrows(axs["arrow"], cutoutSize, 90.0 + self.camRotAngle)
+            drawArrows(axs["arrow"], cutoutSize if cutoutSize > 0 else 30, 90.0 + self.camRotAngle)
```

Comment:
Changing this 30 to a module level constant like `DEFAULT_CUTOUT_SIZE=30` or something like that, so you're not hard-coding it here might be nice, but it's not a big deal.